encoding/thrift/thriftrw-plugin-yarpc/server.go

```diff
@@ -65,6 +65,10 @@ type Interface interface {
 
 <$module := .Module>
 
+type ErrorNamer interface {
+    ErrorName() string
+}
```

Comment:
No need to export this interface. That adds it to the public API of the generated code.

You don't even need to declare it as a named type. You can do,

```
if namer, ok := err.(interface{ ErrorName() string }); ok {
  ...
}
```

But if that feels icky, we can declare the interface in a named, unexported type.

```
type errorNamer interface { ErrorName() string }
```

Reply:
Agree. Will go with the unexported inline declaration though. Thank you!